Ignore non-digest challenges in proxy digest auth

rdn32 opened the pull request:
If HTTPProxyDigestAuth finds itself looking at a non-digest proxy
authentication challenge then it should ignore it rather than fall
over trying to make sense of the challenge's content.

Lukasa commented:
The functionality here seems reasonable to me, though I know that @sigmavirus24 is going to have some comments re testing.

pcreech commented on tests/test_auth.py:
```diff
@@ -75,3 +77,54 @@ def test_handle_407_header_basic(self, extract_cookies_to_jar, proxy_auth_call):
         proxy_auth_call.assert_called_with(req)
+
+
+class ProxyAuthHeaderMatcher(betamax.BaseMatcher):
```
Tests already exist for HTTPProxyDigestAuth in /tests/test_proxy_digest_auth.py.  It might be beneficial to move these to that file

rdn32 replied:
Oops - I didn't notice that file. I'll hold off moving the tests in case more radical changes to them are about to be suggested.

pcreech commented:
Functionality looks good to me, not the best with testing logic yet.
perhaps @sigmavirus24 can give that a glance over?

The failed checks appear to be related to an update in pep8, unrelated to these changes

sigmavirus24 commented:
Yeah if you can update this to do the following:

- Remove the betamax integration tests. Betamax doesn't work at a low-enough level to be useful for this.
- Add unit tests to the file that @pcreech created.

Meanwhile I'll update the toolbelt to fix the pep8 issues. Thanks @rdn32

sigmavirus24 commented:
Nevermind, @pcreech already updated the toolbelt for that pep8 issue.

rdn32 commented:
I made the changes to the tests. I also renamed the existing test method in TestProxyDigestAuth to something more specific to what it is testing, and moved some set-up code that wasn't relevant to the new test methods into it from setUp(). I hope that's OK.

sigmavirus24 commented on tests/test_proxy_digest_auth.py:
```diff
@@ -15,26 +16,88 @@ def setUp(self):
+        first_response = connection.make_response(self.prepared_request)
+        first_response.status_code = 404
+
+        assert not self.auth.last_nonce
```
`assert not` is a bit ... vague. Could this be a bit more specific, e.g.,

```py
assert self.auth.last_nonce is None
```

That's much more explicit.

sigmavirus24 commented on tests/test_proxy_digest_auth.py:
```diff
@@ -15,26 +16,88 @@ def setUp(self):
+        }
+
         # prepared_request headers should be clear before calling auth
         assert not self.prepared_request.headers.get('Proxy-Authorization')
```
I realize I didn't catch that here. Would you mind updating this too?

sigmavirus24 commented on tests/test_proxy_digest_auth.py:
```diff
@@ -15,26 +16,88 @@ def setUp(self):
         # prepared_request headers should be clear before calling auth
         assert not self.prepared_request.headers.get('Proxy-Authorization')
         self.auth(self.prepared_request)
         assert self.prepared_request.headers.get('Proxy-Authorization')
```
Likewise can this be

```py
assert self.prepared_request.headers.get('Proxy-Authorization') is not None
```

sigmavirus24 commented:
I'm truly sorry for the long review cycles @rdn32. I've been very busy lately.
